.github/workflows/pr-actions.yaml

```diff
@@ -236,13 +236,73 @@ jobs:
       overlay-repo: ${{ needs.prepare.outputs.overlay-repo }}
       workspace-path: ${{ needs.prepare.outputs.workspace }}
 
+  add_compatibility_failure_comment:
+    needs:
+      - parse
+      - prepare
+      - checkBackstageCompatibility
+
+    concurrency:
+      group: add_compatibility_failure_comment-${{ github.ref_name }}-${{ github.event.issue.number }}
+      cancel-in-progress: false
+
+    permissions:
+      pull-requests: write
+
+    if: failure() && needs.parse.outputs.command_name == 'publish' && needs.prepare.outputs.overlay-branch != '' && needs.prepare.outputs.workspace != ''
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download compatibility report
+        uses: actions/download-artifact@v4
+        with:
+          name: backstage-compatibility-report
+          path: ./
+
+      - name: Verify compatibility report
+        run: |
+          if [ -f "backstage-compatibility-report.md" ]; then
+            echo "Successfully downloaded backstage-compatibility-report.md"
+          else
+            echo "backstage-compatibility-report.md not found, skipping comment"
+            exit 0
+          fi
+
+      - name: Add compatibility failure comment
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
```

Comment:
Why not add this additional comment text in the `add-publish-completion-comment` job, because it already depends on the `check-backstage-compatibility` job ?
In such a case I guess you would add the `download-compatibility-report` step as the first step of the `add-publish-completion-comment` job.